kview/src/main/java/dev/ikm/komet/kview/mvvm/model/DataModelHelper.java

```diff
@@ -77,4 +104,90 @@ public static Set<ConceptEntity> fetchDescendentsOfConcept(ViewProperties viewPr
                 .collect(Collectors.toSet());
         return allDecendents;
     }
+
+    public static List<PatternEntityVersion> getMembershipPatterns() {
+        List<PatternEntityVersion> membershipPatternList = new ArrayList<>();
+        PrimitiveData.get().forEachPatternNid((patternNid) -> {
+            PatternEntityVersion patternEntityVersion = (PatternEntityVersion)
+                    Calculators.View.Default().stampCalculator().latest(patternNid).get();
+            if (patternEntityVersion.semanticPurposeNid() == TinkarTerm.MEMBERSHIP_SEMANTIC.nid()) {
+                membershipPatternList.add(patternEntityVersion);
+            }
+        });
+        return membershipPatternList;
+    }
+
+    public static boolean isInMembershipPattern(int conceptNid, int patternNid) {
+        int[] semanticCount = PrimitiveData.get().semanticNidsForComponentOfPattern(conceptNid, patternNid);
+        // if the semantic count is empty then it is not a member of that pattern
+        return semanticCount.length > 0;
+    }
+
+    public static void addToMembershipPattern(EntityFacade concept, EntityFacade pattern, ViewCalculator viewCalculator) {
```

Comment:
Can the two arguments be PublicIds?
```java
public static void addToMembershipPattern(PublicId conceptPublicId, PublicId patternPublicId, viewCalculator) {
  // ...
}
```

Reply:
the addToMembershipPattern takes an EntityFacade for 2 reasons: 1) in its method it needs the nid and 2) when it calls createSemantic it needs the publicId... if you want I can pass both the nid and publicId instead of the EntityFacade...